spectroscopy/code_src/mast_functions.py

```diff
@@ -131,13 +131,21 @@ def JWST_get_spec_helper(sample_table, search_radius_arcsec, datadir, verbose,
             print("Nothing to download for source {}.".format(stab["label"]))
             continue
 
-        # Download (supress output)
+        # Download
+        # Observations.download_products prints an info message for every download, like:
+        #   "Downloading URL https://mast... to ./data/... ... [Done]"
+        # Suppress these by redirecting stdout to a trap.
         trap = io.StringIO()
         with redirect_stdout(trap):
             download_results = Observations.download_products(
                 data_products_list_filter, download_dir=this_data_dir)
         if verbose:
             print(trap.getvalue())
+        # [FIXME] Every one of these downloads is failing. What to do?
+        download_results = download_results[download_results["Status"] != "ERROR"]
+        if len(download_results) == 0:
+            print(f"ALL DOWNLOADS FAILED for {stab['label']}")
+            continue
```

Comment:
Before line 145 runs, `download_results` looks like this for the sample labeled 'COSMOS1':

<img width="1200" alt="Screenshot 2024-12-04 at 11 37 09 PM" src="https://github.com/user-attachments/assets/e28628e6-7e18-4780-bd62-c7f0cfbde819">

For copy/paste convenience:
```
Local Path: ./data/JWST/mastDownload/JWST/jw02565-o301_s13239_nirspec_clear-prism/jw02565-o301_s13239_nirspec_clear-prism_x1d.fits
Status: ERROR
Message: HTTPError: 404 Client Error: Not Found for url: https://mast.stsci.edu/api/v0.1/Download/file?uri=mast:JWST/product/jw02565-o301_s13239_nirspec_clear-prism_x1d.fits
URL: https://mast.stsci.edu/api/v0.1/Download/file?uri=mast:JWST/product/jw02565-o301_s13239_nirspec_clear-prism_x1d.fits
```

I'm guessing that the other failure messages are similar but I haven't checked them yet.